TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/util/ActiveContextMonitor.java

```diff
@@ -0,0 +1,263 @@
+    private final ActiveContextTracker globalContextTracker;
+    private final ActiveContextTrackerByActivityType perActivityContextTracker;
+    private final OrderedWorkerTracker<Void> orderedRequestTracker;
+    private final int totalItemsToOutputLimit;
+    private final Function<DiagnosticTrackableCompletableFuture<String,Void>,String> formatWorkItem;
+
+    private final Predicate<Level> logLevelIsEnabled;
+    private final AtomicReference<TreeMap<Duration,Level>> ageToLevelEdgeMapRef;
+
+    public ActiveContextMonitor(ActiveContextTracker globalContextTracker,
+                                ActiveContextTrackerByActivityType perActivityContextTracker,
+                                OrderedWorkerTracker<Void> orderedRequestTracker,
+                                int totalItemsToOutputLimit,
+                                Function<DiagnosticTrackableCompletableFuture<String, Void>, String> formatWorkItem,
+                                Logger logger) {
+        this(globalContextTracker, perActivityContextTracker, orderedRequestTracker, totalItemsToOutputLimit,
+                formatWorkItem, (level, supplier)->logger.atLevel(level).setMessage(supplier).log(),
+                logger::isEnabledForLevel);
+    }
+
+    public ActiveContextMonitor(ActiveContextTracker globalContextTracker,
+                                ActiveContextTrackerByActivityType perActivityContextTracker,
+                                OrderedWorkerTracker<Void> orderedRequestTracker,
+                                int totalItemsToOutputLimit,
+                                Function<DiagnosticTrackableCompletableFuture<String, Void>, String> formatWorkItem,
+                                BiConsumer<Level, Supplier<String>> logger,
+                                Predicate<Level> logLevelIsEnabled) {
+        this(globalContextTracker, perActivityContextTracker, orderedRequestTracker, totalItemsToOutputLimit,
+                formatWorkItem, logger, logLevelIsEnabled, Map.of(
+                        Level.ERROR, Duration.ofSeconds(600),
+                        Level.WARN, Duration.ofSeconds(60),
+                        Level.INFO, Duration.ofSeconds(30),
+                        Level.DEBUG, Duration.ofSeconds(5),
+                        Level.TRACE, Duration.ofSeconds(2)));
+    }
+
+    public ActiveContextMonitor(ActiveContextTracker globalContextTracker,
+                                ActiveContextTrackerByActivityType perActivityContextTracker,
+                                OrderedWorkerTracker<Void> orderedRequestTracker,
+                                int totalItemsToOutputLimit,
+                                Function<DiagnosticTrackableCompletableFuture<String, Void>, String> formatWorkItem,
+                                BiConsumer<Level, Supplier<String>> logger,
+                                Predicate<Level> logLevelIsEnabled,
+                                Map<Level,Duration> levelShowsAgeOlderThanMap) {
```

Comment:
Yes.  I'm using a point-map to represent a a non-duplicative and fully-covering interval map.  That's why I'm using the "olderThan" part and keying off of level rather than time, but I think that the other approaches would be about the same.  (I've never been comfortable using floor/ceiling search operations... even when I understand them, the cognitive load is high)